packages/jest-cli/src/search_source.js

```diff
@@ -255,16 +260,18 @@ export default class SearchSource {
       const tests = searchResult.tests;
 
       // $FlowFixMe: dynamic require.
-      const filter = require(filterPath);
-      const filtered = await filter(tests.map(test => test.path));
+      const filter: Array<FilterResult> = require(filterPath);
```

Comment:
@thymikee seems like `$FlowFixMe` masked the typo :P 
`filter` is a function, not an array, right?

Reply:
ugh, yea, `filterResult.filtered` is an array, `filter` is an async fn :D

Reply:
I got more than a little confused by this now as I converted it to TS 😛 Combination of missing docs and wrong type annotation had me stumped until I `git blame`d and came to this PR